[2025-03-09T05:43:12.580+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-03-09T05:43:12.602+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: submit_spark_job_v2.submit_spark_via_ssh scheduled__2025-03-08T01:00:00+00:00 [queued]>
[2025-03-09T05:43:12.611+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: submit_spark_job_v2.submit_spark_via_ssh scheduled__2025-03-08T01:00:00+00:00 [queued]>
[2025-03-09T05:43:12.611+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-03-09T05:43:12.624+0000] {taskinstance.py:2330} INFO - Executing <Task(SSHOperator): submit_spark_via_ssh> on 2025-03-08 01:00:00+00:00
[2025-03-09T05:43:12.628+0000] {standard_task_runner.py:63} INFO - Started process 7952 to run task
[2025-03-09T05:43:12.632+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'submit_spark_job_v2', 'submit_spark_via_ssh', 'scheduled__2025-03-08T01:00:00+00:00', '--job-id', '32', '--raw', '--subdir', 'DAGS_FOLDER/match_info_spark_process_v2.py', '--cfg-path', '/tmp/tmpgisla60a']
[2025-03-09T05:43:12.634+0000] {standard_task_runner.py:91} INFO - Job 32: Subtask submit_spark_via_ssh
[2025-03-09T05:43:12.697+0000] {task_command.py:426} INFO - Running <TaskInstance: submit_spark_job_v2.submit_spark_via_ssh scheduled__2025-03-08T01:00:00+00:00 [running]> on host 40942f81e2c9
[2025-03-09T05:43:12.774+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.9/site-packages/***/utils/context.py:212 AirflowContextDeprecationWarning: Accessing 'execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_start' or 'logical_date' instead.
[2025-03-09T05:43:12.818+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='submit_spark_job_v2' AIRFLOW_CTX_TASK_ID='submit_spark_via_ssh' AIRFLOW_CTX_EXECUTION_DATE='2025-03-08T01:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-03-08T01:00:00+00:00'
[2025-03-09T05:43:12.819+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-03-09T05:43:12.833+0000] {ssh.py:151} INFO - Creating ssh_client
[2025-03-09T05:43:12.834+0000] {ssh.py:124} INFO - ssh_hook is not provided or invalid. Trying ssh_conn_id to create SSHHook.
[2025-03-09T05:43:12.849+0000] {base.py:84} INFO - Using connection ID 'spark_ssh_conn' for task execution.
[2025-03-09T05:43:12.849+0000] {ssh.py:301} WARNING - No Host Key Verification. This won't protect against Man-In-The-Middle attacks
[2025-03-09T05:43:12.854+0000] {transport.py:1909} INFO - Connected (version 2.0, client OpenSSH_9.6p1)
[2025-03-09T05:43:12.917+0000] {transport.py:1909} INFO - Authentication (publickey) successful!
[2025-03-09T05:43:12.917+0000] {ssh.py:480} INFO - Running command: 
    set -e  # 명령어 실패 시 즉시 종료
    echo "Starting Spark Job on ec2-3-37-216-159.ap-northeast-2.compute.amazonaws.com"

    echo "Submitting Spark Job... for 2025-03-08 data"
   /spark-3.5.3/bin/spark-submit /home/ubuntu/spark_scripts/match_info_pyspark.py 2025-03-08 > spark_job.log 2>&1
    exit_code=$?

    # 실행 로그 출력 (Airflow에서 확인 가능)
    if [ -f spark_job.log ]; then
        echo "=== Spark Job Log Start ==="
        cat spark_job.log
        echo "=== Spark Job Log End ==="
    else
        echo "Warning: spark_job.log not found!"
    fi

    # Spark 실패 감지
    if [ $exit_code -ne 0 ]; then
        echo "Spark Job Failed! Check logs."
        exit 1
    else
        echo "Spark Job Succeeded!"
    fi
    
[2025-03-09T05:43:13.363+0000] {ssh.py:526} INFO - Starting Spark Job on ec2-3-37-216-159.ap-northeast-2.compute.amazonaws.com
[2025-03-09T05:43:13.363+0000] {ssh.py:526} INFO - Submitting Spark Job... for 2025-03-08 data
[2025-03-09T05:43:23.796+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-03-09T05:43:23.811+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/ssh/operators/ssh.py", line 191, in execute
    result = self.run_ssh_client_command(ssh_client, self.command, context=context)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/ssh/operators/ssh.py", line 179, in run_ssh_client_command
    self.raise_for_status(exit_status, agg_stderr, context=context)
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/ssh/operators/ssh.py", line 173, in raise_for_status
    raise AirflowException(f"SSH operator error: exit status = {exit_status}")
airflow.exceptions.AirflowException: SSH operator error: exit status = 1
[2025-03-09T05:43:23.815+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=submit_spark_job_v2, task_id=submit_spark_via_ssh, run_id=scheduled__2025-03-08T01:00:00+00:00, execution_date=20250308T010000, start_date=20250309T054312, end_date=20250309T054323
[2025-03-09T05:43:23.827+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 32 for task submit_spark_via_ssh (SSH operator error: exit status = 1; 7952)
[2025-03-09T05:43:23.857+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-03-09T05:43:23.880+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-03-09T05:43:23.882+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
